machine/quanta/quanta_common_rangeley/rootconf/sysroot-init/rtc-sync.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+RTC_DS1339_I2C_SLAVE=0x68
+
+. /lib/onie/functions
+
+/usr/bin/i2cget -y 0 $RTC_DS1339_I2C_SLAVE &> /dev/null && {
+	log_info_msg "Initializing RTC DS1339 ..."
+	mkdir /sysfs
+	mount -t sysfs sysfs /sysfs
+	/bin/echo ds1339 0x68 > /sysfs/devices/pci0000:00/0000:00:1f.3/i2c-0/new_device
+	umount /sysfs
+	rm -rf /sysfs
+	/bin/mknod /dev/rtc1 c 254 1
+	/bin/ln -sf /dev/rtc1 /dev/rtc
+	/sbin/hwclock --hctosys
+	log_info_msg "Initializing RTC DS1339 Done ..."
+}
```

Comment:
That is a really unfortunate hardware design for x86.

This puts additional requirements on the kernel as the RTC *should* be properly configured into the kernel and the RTC programmed by the *kernel* at boot time.  Programming the RTC from user space after the system as booted is a pretty big hack.

For example the initial dmesg log messages will have the wrong date/time before the RTC is fixed up from userspace.

Please encourage your hardware team to do it the standard x86 way in the future.